### ISSUE TITLE
exp(-Inf) yields NaN

### ISSUE THREAD

**Issue by saschatimme** [JuliaMath/DoubleFloats.jl#86]
```julia
julia> using DoubleFloats

julia> exp(-Inf)
0.0

julia> exp(-Double64(Inf))
NaN

julia> exp(-Inf+0.0im)
0.0 + 0.0im

julia> exp(ComplexDF64(-Inf))
NaN - NaN*im
```

**Comment by JeffreySarnoff** [comment]
stems from this bug -- fixing (thank you)
```
julia> -Double64(Inf)
NaN
```

**Comment by JeffreySarnoff** [comment]
There is a similar problem with other elementary functions. Fixing.

**Comment by JeffreySarnoff** [comment]
fixed in current master, merge of v0.9.10 pending
